### PR TITLE
fix undefined constants.

### PR DESCRIPTION
**Pull request by SebSept**
In fact it only hides the errors, but it do the jobs (it seems).

| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Avoid undefined constants : makes phpstan tests fail on CI
| Type?             | bug fix 
| BC breaks?        |  no
| Deprecations?     | no
| Fixed ticket?     | #53 
| How to test?      | run the analyse , ?(use php 7.4)
| Possible impacts? |  may hide problems inside Prestashop, but this is not to test Prestashop.

### CODE REVIEW

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
I'm not sure it's the right way to handle it, here, you're not fixing the undefined constant, you just hide them :thinking:

**Comment by SebSept** [comment]
For sure it's an hard and ugly solution.
defining the missing constants is way better.

But if I test against old version of Prestashop, 1.7.5.2 for example, there is another error running on php8.
```
» _PS_ROOT_DIR_=/home/http/tests/prestashop/PrestaShop-1.7.5.x php80 ./phpstan.phar analyse ./
Note: Using configuration file /home/http/tests/phpstan-prestashop/blockreassurance/phpstan.neon.
PHP Fatal error:  Array and string offset access syntax with curly braces is no longer supported in /home/http/tests/prestashop/PrestaShop-1.7.5.x/vendor/icanboogie/common/lib/helpers.php on line 306
```
This error is not solved by adding constants.

But it's solved with the @ trick ... 

That's not a problem for me, I won't test old versions against php8, So this error may not happen on real use cases.. Not for me at least.

Maybe we can define the constants and forget about the rest ... 🤔

**Comment by PierreRambaud** [comment]
> For sure it's an hard and ugly solution.
> defining the missing constants is way better.
> 
> But if I test against old version of Prestashop, 1.7.5.2 for example, there is another error running on php8.
> 
> ```
> » _PS_ROOT_DIR_=/home/http/tests/prestashop/PrestaShop-1.7.5.x php80 ./phpstan.phar analyse ./
> Note: Using configuration file /home/http/tests/phpstan-prestashop/blockreassurance/phpstan.neon.
> PHP Fatal error:  Array and string offset access syntax with curly braces is no longer supported in /home/http/tests/prestashop/PrestaShop-1.7.5.x/vendor/icanboogie/common/lib/helpers.php on line 306
> ```
> 
> This error is not solved by adding constants.
> 
> But it's solved with the @ trick ...
> 
> That's not a problem for me, I won't test old versions against php8, So this error may not happen on real use cases.. Not for me at least.
> 
> Maybe we can define the constants and forget about the rest ...

You are using php8, PrestaShop is not compatible with it yet :)

**Comment by SebSept** [comment]
Sure and I did not even start to test my modules with php8.
I just saw this error on a github action.

**Review by SebSept — COMMENTED** [review]
that's fixed

File: phpstan/bootstrap.php [hunk 1]
```diff
@@ -32,6 +32,8 @@ function requireFileIfItExists(string $filepath): bool
 // Add PrestaShop composer autoload
 define('_PS_ADMIN_DIR_', $rootDir . '/admin-dev/');
 define('PS_ADMIN_DIR', _PS_ADMIN_DIR_);
+define('_THEME_NAME_', 'default-bootstrap');
```
Comment (PierreRambaud) [inline phpstan/bootstrap.php:35]: If you do it, you can remove the line 122

**Comment by SebSept** [comment]
replaced by #59

**Pull request closed without merging**